Add Pluracle contract

AugustoL opened the pull request.

AugustoL committed:
Add MediumPriceOracle contract

jpkcambridge commented on contracts/MediumPriceOracle.sol:
```diff
@@ -0,0 +1,79 @@
+ * @title UintPluracle
+ * @dev An oracle of uint256 types oracles that provides a medium price.
+ */
+contract UintPluracle is Ownable {
```
Should we make this consistent with the title of the file?

AugustoL replied:
yes, changing it !

AugustoL committed:
Rename MediumPrice to UintPluracle

jpkcambridge commented on contracts/UintPluracle.sol:
```diff
@@ -0,0 +1,79 @@
+  uint256 public _lastTimestamp;
+
+  function UintPluracle(
+    uint256 reward, string dataType
```
Is `dataType` getting used?

jpkcambridge commented on contracts/UintPluracle.sol:
```diff
@@ -0,0 +1,79 @@
+    // Get all prices
+    uint256 totalPrice;
+    uint256 totalOracles;
+    for(uint8 i = 0; i < oracles.length; i ++) {
```
Should we make this uint256?

jpkcambridge committed:
Updateis to uintPluracle
Add maximumUpdateFrequency to protect against reward draining
Add dataType check on oracles to make sure averaging can be done
